### PR TITLE
Receipt thumbnail should show the top of the receipt

### PR DESCRIPTION
**Pull request by dukenv0307**
### Fixed Issues

$ https://github.com/Expensify/App/issues/34120
PROPOSAL: https://github.com/Expensify/App/issues/34120#issuecomment-1882437734

### Tests

1. Create an IOU using receipt scanning
2. Observe the thumbnail of the receipt
3. Verify that Receipt thumbnail shows the top of the receipt

- [x] Verify that no errors appear in the JS console

### QA Steps

1. Create an IOU using receipt scanning
2. Observe the thumbnail of the receipt
3. Verify that Receipt thumbnail shows the top of the receipt

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/Expensify/App/assets/129500732/c9fa5d15-f6e7-4ffd-a8c5-90fca8dfb8af

</details>

<details>
<summary>Android: mWeb Chrome</summary>

android-mweb

</details>

<details>
<summary>iOS: Native</summary>

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/Expensify/App/assets/129500732/cc5d2f2e-5002-4505-8720-fb04b6fa9b35

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/129500732/b6e9c22c-df05-4ae5-94b6-cd710d13acf4

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/Expensify/App/assets/129500732/2cbb1c15-e6ec-462d-b06e-145d9dcb75e3

</details>

### CODE REVIEW

**Comment by dragnoir** [comment]
it's not working:

image

**Comment by cubuspl42** [comment]
@dukenv0307 I can see that you're actively working on this! What's the current status of the PR? I think it would be great to prioritize this highly.

@dragnoir Thanks for volunteering to test this PR, but it's not needed or welcome. We'll ensure to test all image ratios.

**Comment by cubuspl42** [comment]
> @cubuspl42 The PR is ready for review. Have a question here, should we always set `aspectRaito: 1` for wide image (width > height) or only set for the case of stack images (`ReportActionItemImages`).

Are you discussing something that affects user-observable behavior? If so, could you share screenshots of both options?

**Comment by cubuspl42** [comment]
It's a good question. I believe we should stop testing with moon images and figure out what might be the actual user stories here.

Let's consider this horizontal picture of a recipe (4:3):

walmart_receipe_horizontal

Also 16:9:

walmart_receipe_horizontal_wide

We shouldn't focus on anything wider. It's already focusing on corner cases.

(it's digitally edited, I didn't have a horizontal recipe image handy)

**Comment by cubuspl42** [comment]
The linter is still complaining

**Comment by dragnoir** [comment]
@cubuspl42 sorry about that. I was just wondering about the aspect ratio when there's no parent width/height specified. You're doing great with your work, and I'm keeping an eye on it to learn different ways to tackle this.

**Comment by cubuspl42** [comment]
@dukenv0307 Ouch; conflicts

**Comment by cubuspl42** [comment]
I believe things are looking good both for vertical and horizontal images.

image

image

**Comment by cubuspl42** [comment]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos

<details>
<summary>Android: Native</summary>

image

</details>

<details>
<summary>Android: mWeb Chrome</summary>

image

</details>

<details>
<summary>iOS: Native</summary>

image

</details>

<details>
<summary>iOS: mWeb Safari</summary>

image

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

image

</details>

<details>
<summary>MacOS: Desktop</summary>



</details>

File: src/components/Image/BaseImage.tsx [hunk 1]
```diff
@@ -13,14 +14,23 @@ function BaseImage({onLoad, ...props}: WebImageProps & BaseImageProps) {
...
             const {width, height} = event.nativeEvent.source;
             onLoad({nativeEvent: {width, height}});
+
+            if (objectPositionTop) {
+                if (width > height) {
+                    setAspectRatio(1);
+                    return;
+                }
+                setAspectRatio(height ? width / height : 'auto');
+            }
```
Comment (cubuspl42) [inline src/components/Image/BaseImage.tsx:24]: We shouldn't duplicate this code.

Can't we move the feature implementation to `Image`?

**Review by jakub-trzebiatowski — CHANGES_REQUESTED** [review]

File: src/components/Image/BaseImage.native.tsx [hunk 2]
```diff
@@ -13,14 +14,23 @@ function BaseImage({onLoad, ...props}: ExpoImageProps & BaseImageProps) {
...
         },
-        [onLoad],
+        [onLoad, objectPositionTop],
     );
 
     return (
         <ExpoImage
             // Only subscribe to onLoad when a handler is provided to avoid unnecessary event registrations, optimizing performance.
             onLoad={onLoad ? imageLoadedSuccessfully : undefined}
+            style={[style, !!aspectRatio && {aspectRatio, height: 'auto'}, objectPositionTop && !aspectRatio && {opacity: 0}]}
```
[the remark this answers is not in the document]
Comment (cubuspl42) [reply src/components/Image/BaseImage.native.tsx]: You're right, sorry. Still...

```ts
aspectRatio ? {aspectRatio, height: 'auto'} : (objectPositionTop && {opacity: 0})
```

File: src/components/ReportActionItem/ReportActionItemImage.tsx [hunk 3]
```diff
@@ -64,14 +64,18 @@ function ReportActionItemImage({thumbnail, image, enablePreviewModal = false, tr
...
-                source={{uri: thumbnail ?? image}}
-                style={[styles.w100, styles.h100]}
-            />
+            <View style={[styles.w100, styles.h100]}>
+                <Image
+                    source={{uri: thumbnail ?? image}}
+                    style={[styles.w100, styles.h100]}
+                    objectPositionTop
+                />
+            </View>
```
Comment (cubuspl42) [inline src/components/ReportActionItem/ReportActionItemImage.tsx:78]: Hm, what happened here and why?

File: src/components/Image/index.js [hunk 4]
```diff
@@ -28,11 +30,30 @@ function Image({source: propsSource, isAuthTokenRequired, session, ...forwardedP
         // eslint-disable-next-line react-hooks/exhaustive-deps
     }, [propsSource, isAuthTokenRequired]);
 
+    const imageLoadedSuccessfully = useCallback(
```
Comment (cubuspl42) [inline src/components/Image/index.js:33]: Hm, I don't think that a name like _"something happened"_ is a conventional way to name a function. Typically it's _"do something"_ or _"on event happened"_ (for event handlers).

Another topic is that we have an internal naming convention **not to** name event handlers _"on something happened"_; Expensify states we should always prefer _"do something"_. I don't agree 100% with this convention, but it's not up for debate.

I believe that an escape path from this situation can be like this:

1. Extract...

```ts
const updateAspectRatio = useCallback(
    (width, height) => {
        if (objectPositionTop) {
            if (width > height) {
                setAspectRatio(1);
                return;
            }
            
            setAspectRatio(height ? width / height : 'auto');
        }
    },
    [objectPositionTop],
);
```

2. Rename `imageLoadedSuccessfully` to `handleLoad` and change the implementation...

```ts
const handleLoad = useCallback(
    (event) => {
        const {width, height} = event.nativeEvent;

        onLoad(event);

        updateAspectRatio(width, height);
    },
    [onLoad],
);
```

This way, everything is named according to what we do, and we stick both to common sense (kind of) and to Expensify naming rules.

File: src/components/Image/index.js [hunk 5]
```diff
@@ -28,11 +30,30 @@ function Image({source: propsSource, isAuthTokenRequired, session, ...forwardedP
...
+        (event) => {
+            const {width, height} = event.nativeEvent;
+
+            onLoad(event);
+
+            if (objectPositionTop) {
+                if (width > height) {
+                    setAspectRatio(1);
+                    return;
```
Comment (cubuspl42) [inline src/components/Image/index.js:42]: Can we change this logic to if-else?

**Comment by cubuspl42** [comment]
I gave it one more review round. We're very close.

**Comment by shawnborton** [comment]
Can you please tag @Expensify/design for review once this is ready to go as well? Thanks!

**Comment by shawnborton** [comment]
Okay cool, that looks pretty good to me. One thing I wanted to check on was a weird behavior I observed in a different video somewhere (thanks @trjExpensify for sharing) where the receipt image kind of jumped/zoomed after it loaded:

https://github.com/Expensify/App/assets/141406735/c0a50c7c-7c17-4002-824a-3c74488ba794

Just want to make sure that's not happening here!

**Comment by cubuspl42** [comment]
I bumped folks in the PR we're holding for: https://github.com/Expensify/App/pull/36560#issuecomment-1961201403

**Comment by cubuspl42** [comment]
Have you uploaded the actual images returned by the backend? It doesn't look so. I'm not sure what I'm actually looking at.

It would be good to compare...

- the original uploaded photo
- the in-context screenshot of a thumbnail before the backend response (like [this](https://private-user-images.githubusercontent.com/2590174/301493119-041387a7-147a-468c-95d1-cf2ee9090430.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDg2OTAyNDksIm5iZiI6MTcwODY4OTk0OSwicGF0aCI6Ii8yNTkwMTc0LzMwMTQ5MzExOS0wNDEzODdhNy0xNDdhLTQ2OGMtOTVkMS1jZjJlZTkwOTA0MzAucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDIyMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAyMjNUMTIwNTQ5WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NTZiYjlhNjBhM2NjYmJlZmUxOTNmYjQyZDZlYzVhNDRlNDRjNWMwOWQ4YmYzNzYwMTA1ZWI2MDdiNWUxNjExYSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.WrXKmCvw4_8npmpJ22-gthxr2EakgackOeo6sCA7ck4)) 
- the actual thumbnail image returned by the backend
- analogically, the in-context screenshot of a thumbnail **after** the backend response

Speaking about the action points, we could...
- Adjust the backend thumbnail generation
- Adjust the frontend presentation of the backend-returned thumbnail
- Accept the behavior, if it occurs only for rare-in-practice ratios

I'm not sure which is the best way forward yet, as I don't understand the problem in full.

**Comment by shawnborton** [comment]
I think ideally there wouldn't be a difference at all in what you see regarding the thumbnail during the various phases between upload to then processing the expense. I'm not sure from a technical standpoint how we pull that off, but just noting that ideally we can keep the image view as consistent as possible.

**Comment by cubuspl42** [comment]
@shawnborton I agree. I just sketched possible options. 100% consistent behavior is much preferred.

First, we need to debug and understand the problem.

**Comment by cubuspl42** [comment]
We'd like to achieve this, like Shawn said:

> I think ideally there wouldn't be a difference at all in what you see regarding the thumbnail during the various phases between upload to then processing the expense. I'm not sure from a technical standpoint how we pull that off, but just noting that ideally we can keep the image view as consistent as possible.

Could you intercept the literal image returned by the backend for debugging purposes, if you agree that it would be helpful?

I'm just trying to suggest some steps here, but what we actually want is choosing a way forward, like I mentioned before. As the ratio of the image is quite "normal", I'd say we choose from...

- Adjust the backend thumbnail generation
- Adjust the frontend presentation of the backend-returned thumbnail

We're responsible for solving the issue holistically. It can mean either solving this on the frontend ourselves **or** getting in an agreement with internal engineers that this should be solved on the backend.

I don't want to force a particular way to you. You can either continue investigating / debugging this situation from the frontend perspective or start a discussion on Slack, discussing the potential backend changes. You can try tagging [Alex Beaman](https://github.com/Beamanator), as he's the assigned internal engineer to our GH issue.

**Comment by cubuspl42** [comment]
Was it also "weird" before our PR changes? Was the "jump / zoom" glitch we're observing also happening? If it did, did we made the "jump / zoom" glitch noticeably worse?

**Comment by cubuspl42** [comment]
@dukenv0307 https://github.com/Expensify/App/pull/36560 got merged

File: src/components/Image/index.tsx [hunk 6]
```diff
@@ -1,11 +1,39 @@
...
+import React, {useCallback, useMemo, useState} from 'react';
 import {withOnyx} from 'react-native-onyx';
 import CONST from '@src/CONST';
 import ONYXKEYS from '@src/ONYXKEYS';
 import BaseImage from './BaseImage';
-import type {ImageOnyxProps, ImageOwnProps, ImageProps} from './types';
+import type {ImageOnLoadEvent, ImageOnyxProps, ImageOwnProps, ImageProps} from './types';
 
-function Image({source: propsSource, isAuthTokenRequired = false, session, ...forwardedProps}: ImageProps) {
+function Image({source: propsSource, isAuthTokenRequired = false, session, onLoad, objectPositionTop, style, ...forwardedProps}: ImageProps) {
```
Comment (cubuspl42) [inline src/components/Image/index.tsx:8]: What do you think about changing `objectPositionTop: boolean` to `objectPosition: 'top' | 'initial'`? The whole thing has been inspired by CSS from the beginning, so maybe we can make the one last step to model this similarly to how CSS does and also analogously to many inline styles.

`initial` would be the default and would trigger the default behavior. [I took the name from CSS](https://developer.mozilla.org/en-US/docs/Web/CSS/initial).

Sorry about not sugesting this earlier, this just came to my mind.

**Review by jakub-trzebiatowski — CHANGES_REQUESTED** [review]

File: src/components/Image/index.tsx (hunk 6 above)
Comment (cubuspl42) [reply src/components/Image/index.tsx:8]: Great, let's do it 👍

**Comment by cubuspl42** [comment]
Hm, is this working at all?

https://github.com/Expensify/App/assets/2590174/0d2914d7-47c1-4b90-91ce-3301a0940a49

**Comment by cubuspl42** [comment]
I created a new workspace to make things as reproducable as possible. If in doubt, you can create a new workspace too.

This time, I used a horizontal image, but id didn't change much. The top of the image is only shown after a spinner/refresh.

https://github.com/Expensify/App/assets/2590174/ec273dbe-01ee-4032-8aad-d2d476f144c9

**Comment by cubuspl42** [comment]
Thanks, got it! Yes, we shouldn't call this "offline" to avoid confusion with the case when my Internet connection is down.

I'll give it another testing round in a few moments

**Comment by cubuspl42** [comment]
Conflicts 🙁 Sorry about that

File: src/components/ReceiptImage.tsx [hunk 7]
```diff
@@ -115,10 +116,11 @@ function ReceiptImage({
             <ThumbnailImage
                 previewSourceURL={source ?? ''}
                 style={[styles.w100, styles.h100]}
-                isAuthTokenRequired
+                isAuthTokenRequired={isAuthTokenRequired ?? false}
```
Comment (cubuspl42) [inline src/components/ReceiptImage.tsx:119]: Could you quickly comment on the changes related to the auth token? (this line, but also some other lines in the whole PR diff)

Does adjusting the thumbnail positioning have some relation with auth tokens? I'm trying to ensure we got conflict resolution right.

Comment (cubuspl42) [reply src/components/ReceiptImage.tsx:119]: Sorry, I forgot. This is definitely a sign we should wrap this up.

**Comment by cubuspl42** [comment]
https://github.com/Expensify/App/assets/2590174/ce785b6b-6485-4ee9-baa7-5cda95533b25

This is not top 🤔

image

**Comment by cubuspl42** [comment]
@dukenv0307 I'm not sure, I'm checking again.

**Comment by cubuspl42** [comment]
I used this receipt:

<details>

walmart_receipt

</details>

I created a fresh workspace and created a scan money request. I waited till the image reloaded. You can try doing the exact same steps with the file I provided.

https://github.com/Expensify/App/assets/2590174/4a0265bb-cf54-408d-ae03-b37bd724c364

**Comment by cubuspl42** [comment]
Oh, so it's literally cropped? Interesting, does it happen for all vertical images?

I wonder if we need to coordinate with the backend team, so there's no "jump" during loading. The user doesn't care which part is handled by the backend and which part is handled by the frontend.

**Comment by cubuspl42** [comment]
Okey, I'll re-test it with normal ratios! Real users will likely only upload photos from the camera, uncropped.

**Comment by cubuspl42** [comment]
> tested and it only happens for scan requests with very long vertical images. The scan split doesn't happen that.

This is not what I observe.

As I observe it, it appears that the backend returns a square thumbnail even when I upload a standard, 4:3 vertical image.

Image I used:

<details>

walmart_receipt_vertical

</details>

Thumbnail I got from the backend:

<details>

receipt_thumbnail

</details>

Images aligned on top of each other:

image

This is just basic, centered cropping.

**Comment by cubuspl42** [comment]
@Beamanator Would it be easy to adjust the backend, so it chooses the top portion of the image when generating the recipe thumbnail?

I'm also wondering if it's easy/possible to distinguish between the recipes and other images, and whether we're fine with aligning to the top of all user-uploaded images 🤔

**Comment by cubuspl42** [comment]
@Beamanator Thanks!

Just as a reminder of the current state of this issue/PR:

We're implementing a feature [originally discussed on Slack](https://expensify.slack.com/archives/C049HHMV9SM/p1704757554079199) (it was called "bug", but this never worked "as expected", as far as I understand) to position the receipt thumbnails so the top of the image is presented, not the center.

We were holding on another issue for quite some time.

Currently, we are (mostly?) done with the frontend part of the task, but it seems that the backend behavior also needs some changes, as the thumbnail image comes pre-cropped to the frontend.

**Comment by Beamanator** [comment]
Ok I do have a theory on this one, it looks like a few months back (early feb) we merged this PR - https://github.com/Expensify/Web-PDFs/pull/560/files - which had the intent to fix https://github.com/Expensify/App/issues/32649

We mayyyy have added an unnecessary `center` command which could be why the thumbnails are always centered on the image 🤔

**Comment by Beamanator** [comment]
@cubuspl42 will you please retest so we can hopefully get this merged today? 🙏

**Comment by cubuspl42** [comment]
I have trouble testing this right now, the scan requests... disappear for me. Some backend hiccups?

**Comment by Beamanator** [comment]
That definitely doesn't sound good @cubuspl42 ! Is it reproducible? Can you report a bug if it is?

**Comment by cubuspl42** [comment]
I'm sorry for a lag, I'll retest this today.

**Comment by cubuspl42** [comment]
I had really weird issues when testing this, but I believe that they were on the backend and unrelated to this PR.

Things look good now, no jump after the refresh!

https://github.com/Expensify/App/assets/2590174/031fa97b-9ded-4a99-b21e-52f15afad7b5

**Review by jakub-trzebiatowski — APPROVED** [review]

**Comment by Beamanator** [comment]
Thanks for retesting everything @cubuspl42 ! Do you mind doing 1 more quick retest since @dukenv0307 just merged main? (since it had been a while) 🙏

**Comment by cubuspl42** [comment]
@Beamanator I always merge `main` on my end before any testing

**Review by Beamanator — APPROVED** [review]
LGTM

**Comment by kbecciv** [comment]
This PR is failing because of issue(https://github.com/Expensify/App/issues/34120)

The issue is reproducible in: All environments 


https://github.com/Expensify/App/assets/93399543/67526557-8146-4086-87e8-fd8fe74ce117


https://github.com/Expensify/App/assets/93399543/34eed6be-d5bb-4c62-91ab-52c3683d04e8


https://github.com/Expensify/App/assets/93399543/41849363-9ce4-4457-b4a3-4776684dbe22


https://github.com/Expensify/App/assets/93399543/d31418e4-9147-402a-a2a5-54608dcfc9a9


https://github.com/Expensify/App/assets/93399543/6277b291-90e3-4511-94e1-413e1e5d1ebd

**Comment by cubuspl42** [comment]
These are two different kinds of previews; in the videos, we can see that the ones we worked on (report action previews) **do** align to the top: 

image

I recall that we discussed this other kind of previews at some point, but I can't find it now.

@dukenv0307

Do you have some more input here? Also, could you check if covering this second kind of previews is challenging technically? Does it also need backend changes?

**Comment by cubuspl42** [comment]
I think that that's what we're talking about. I think the users, mentally, don't consider various previews (submit expense previews vs. report action previews) to be really different, and they intuitively expect consistency.

Yes, please create a follow-up!

**Pull request merged**